### PR TITLE
[Doc] Files() supports NAS

### PR DESCRIPTION
**Pull request by EsoragotoSpirit**
Signed-off-by: EsoragotoSpirit <wanglichen@starrocks.com>

## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.4
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0

### CODE REVIEW

File: docs/en/sql-reference/sql-functions/table-functions/files.md [hunk 1]
```diff
@@ -669,3 +704,23 @@ FILES(
...
+
+```SQL
+-- CSV
+INSERT INTO FILES(
+  'path' = 'file:///home/ubuntu/csvfile/', 
+  'format' = 'csv', 
+  'csv.column_separator' = ',', 
+  'csv.row_delimitor' = '\n'
+)
+SELECT 1 AS a, "csv" AS b;
```
Comment (jaogoy) [inline docs/en/sql-reference/sql-functions/table-functions/files.md:718]: Use the same select statement as above `SELECT * FROM sales_records`?

File: docs/en/sql-reference/sql-functions/table-functions/files.md [hunk 2]
```diff
@@ -669,3 +704,23 @@ FILES(
 )
 SELECT * FROM sales_records;
 ```
+
+Unload the query results into CSV and Parquet files in NFS(NAS):
+
+```SQL
+-- CSV
+INSERT INTO FILES(
```
Comment (jaogoy) [inline docs/en/sql-reference/sql-functions/table-functions/files.md:712]: at line 695:
Unify the statement as `INSERT INTO FILES(` (as a single line)

File: docs/en/sql-reference/sql-functions/table-functions/files.md (hunk 1 above)
Comment (jaogoy) [reply docs/en/sql-reference/sql-functions/table-functions/files.md:718]: Keep the same indent as 4 space.

File: docs/en/unloading/unload_using_insert_into_files.md [hunk 3]
```diff
@@ -201,6 +201,30 @@ FILES(
...
+
+```SQL
+-- Unload data into CSV files.
+INSERT INTO FILES(
+  'path' = 'file:///home/ubuntu/csvfile/', 
+  'format' = 'csv', 
+  'csv.column_separator' = ',', 
+  'csv.row_delimitor' = '\n'
+)
+SELECT 1 AS a, "csv" AS b;
```
Comment (jaogoy) [inline docs/en/unloading/unload_using_insert_into_files.md:218]: Use `SELECT * FROM sales_records`

**Review by jaogoy — APPROVED** [review]

**Review by wangsimo0 — APPROVED** [review]

**Comment by EsoragotoSpirit** [comment]
https://github.com/Mergifyio backport branch-3.3

**Pull request merged**